src/index.test.ts

```diff
@@ -176,6 +176,13 @@ describe('basic retry and timeout usage', () => {
     expect(res).toEqual(success(2));
   });
 
+  it('retries and resolves successful asynchronous functions with varying timeouts', async () => {
+    jest.spyOn(operations, 'successFn');
+    const res = await go(operations.successFn, { attemptTimeoutMs: [5, 10, 15, 25], retries: 3 });
+    expect(operations.successFn).toHaveBeenCalledTimes(4);
```

Comment:
This isn't actually testing the the timeout delays - it would need to do some kind of time comparison. Something like this is probably needed:

https://github.com/api3dao/promise-utils/blob/e34a523deabf3520146c6a622a7a09dc0d238779/src/index.test.ts#L756-L762